core/state_processor.go

```diff
@@ -744,26 +744,26 @@ func applyTransaction(msg types.Message, parent *types.WorkObject, config *param
 		return nil, nil, err
 	}
 
-	var ETXRCount int
-	var ETXPCount int
+	var ETXRGas uint64
+	var ETXPGas uint64
 	for _, tx := range result.Etxs {
 		// Count which ETXs are cross-region
 		if tx.To().Location().CommonDom(nodeLocation).Context() == common.REGION_CTX {
-			ETXRCount++
+			ETXRGas += tx.Gas()
 		}
 		// Count which ETXs are cross-prime
 		if tx.To().Location().CommonDom(nodeLocation).Context() == common.PRIME_CTX {
-			ETXPCount++
+			ETXPGas += tx.Gas()
```

Comment:
should it be gas used or gas limit of the transaction that counts ?

Reply:
We don't know  how much gas an ETX will use until it's executed in the destination. We just add the gas provided, with the assumption that it will all be used (except in the Qi->Quai conversion case, we only count 100k gas)

Reply:
okay makes sense